google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/util/AppEngineUtil.java

```diff
@@ -163,13 +159,61 @@ private static boolean isFlexCompat(@NotNull Project project,
         if (root != null) {
           XmlTag vmTag = root.findFirstSubTag("vm");
           if (vmTag != null) {
-            isVmTrue = Boolean.parseBoolean(vmTag.getValue().getTrimmedText());
+            return vmTag;
+          } else {
+            return root.findFirstSubTag("env");
           }
         }
       }
     }
 
-    return isVmTrue;
+    return null;
+  }
+
+  /**
+   * Determines if a project is set up like an App Engine standard project but is configured
+   * in 'compatibility' mode. This indicates that the project runs in the flexible environment.
+   *
+   * <p>A flex compat project has an appengine-web.xml with either:
+   * {@code
+   * <vm>true</vm>
+   * <env>flex</env>
+   * }
+   */
+  public static boolean isFlexCompat(@NotNull Project project, @NotNull DeploymentSource source) {
+    return isFlexCompat(project, getArtifact(source));
+  }
+
+  private static boolean isFlexCompat(@NotNull Project project,
+      @Nullable Artifact artifact) {
+    if (artifact == null) {
+      return false;
+    }
+
+    XmlTag compatConfig = getFlexCompatXmlConfiguration(project, artifact);
+
+    if (compatConfig == null) {
+      return false;
+    }
+
+    String tagName = compatConfig.getName();
+
+    if ("vm".equals(tagName)) {
```

Comment:
yea, its a little different but it is also changing the value of the tag names. I'll think about ways to structure this better.